src/services/ledger/index.types.d.ts

```diff
@@ -36,29 +36,30 @@ type AddOnchainSendLedgerMetadata = NonIntraledgerLedgerMetadata & {
 type AddColdStorageLedgerMetadata = NonIntraledgerLedgerMetadata & {
   hash: OnChainTxHash
   payee_addresses: OnChainAddress[]
-  currency: WalletCurrency
+  currency?: WalletCurrency
 }
 
 type AddColdStorageReceiveLedgerMetadata = AddColdStorageLedgerMetadata
 
 type AddColdStorageSendLedgerMetadata = AddColdStorageLedgerMetadata
 
-type IntraledgerLedgerMetadata = LedgerMetadata & {
-  memoPayer: string | undefined
-  username: Username | undefined
+type IntraledgerBaseMetadata = LedgerMetadata & {
+  usd: DisplayCurrencyBaseAmount // to be renamed amountDisplayCurrency
+  memoPayer: string | undefined | null
```

Comment:
I think we decided to standardise on `undefined` instead of `null`... can we fix this when its constructed or is there a reason its like this?
Also why not `memoPayer?: string` instead of the `undefined`

Reply:
I've switched to using undefined.